pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -122,6 +123,33 @@ class CreateVM extends React.Component {
             break;
         case 'source':
             this.setState({ [key]: value });
+
+            // Clears the previously set timer.
+            clearTimeout(this.typingTimeout);
+
+            if (this.state.sourceType == URL_SOURCE && this.state.vendor == NOT_SPECIFIED) {
+                const autodetectOSHandler = (os) => autodetectOS(os)
+                        .then(res => {
+                            const osName = res
+                                    .split('for OS')
+                                    .pop()
+                                    .replace(/['"\n]+/g, '')
+                                    .trim();
+                            const osEntry = this.props.osInfoList.filter(osEntry => osEntry.name == osName);
+
+                            if (osEntry) {
+                                this.setState({
+                                    vendor: osEntry[0].vendor,
+                                    os: osEntry[0].shortId
+                                });
+                                notifyValuesChanged('vendor', osEntry[0].vendor);
+                                notifyValuesChanged('os', osEntry[0].shortId);
+                            }
+                        });
+
+                // Reset the timer, to make the http call after 250MS
+                this.typingTimeout = setTimeout(() => autodetectOSHandler(value), 250);
```

Comment:
Completely optional, but in pkg/lib/cockpit-components-file-autocomplete.jsx we now use the throttle-debounce module (see commit d53c40350d7d67). That might simplify things a bit.

Reply:
I don't have preference. I adjusted the code to use debounce.